optuna/integration/allennlp.py

```diff
@@ -15,12 +18,166 @@
     import allennlp
     import allennlp.commands
     import allennlp.common.util
-    from allennlp.training import EpochCallback
 
+# EpochCallback is conditionally imported because allennlp may be unavailable in
+# the environment that builds the documentation.
 if _imports.is_successful():
     import _jsonnet
+    from allennlp.training import EpochCallback
 else:
-    EpochCallback = object  # NOQA
+    # I disable mypy here since `allennlp.training.EpochCallback` is a subclass of `Registrable`
+    # (https://docs.allennlp.org/master/api/training/trainer/#epochcallback) but `EpochCallback`
+    # defined here is not `Registrable`, which causes a mypy checking feailure.
+    class EpochCallback:  # type: ignore
+        """Stub for EpochCallback."""
+
+        @classmethod
+        def register(cls: Any, *args: Any, **kwargs: Any) -> Callable:
+            """Stub method for `EpochCallback.register`.
+
+            This method has the same signature as
+            `Registrable.register <https://docs.allennlp.org/master/
+            api/common/registrable/#registrable>`_ in AllenNLP.
+
+            """
+
+            def wrapper(subclass: Any, *args: Any, **kwargs: Any) -> None:
+                return subclass
+
+            return wrapper
+
+
+PPID = os.getppid()
```

Comment:
How about making this constant and the ones below private as we did before, by prefixing them with underscores?

Reply:
7b2c92f made `PPID` private. :pray: